tests/testthat/test-word.R

```diff
@@ -0,0 +1,10 @@
+test_that("process_markdown() correctly resolves reference", {
+  skip_if_not_pandoc()
+  # testing using markdown_document2() for snapshotting easier
+  rmd <- local_rmd_file(
+    "# Theory {#theory}", "", "see \\@ref(label1)", "",
+    "## Some other header {#label1}", "", "Content"
```

Comment:
Should longer text be used here? Otherwise the `--wrap` option probably has no effect.

Reply:
It seems like it is enough to recreate the bug as the HTML will be longer. The wrapping happens at 72 (default value for `--columns`). 

``` r
# 4 lines
pandoc::pandoc_convert(text = "## Some other header {#label1}", to = "html4", args = c("--section-divs", "--number-sections"), version = "2.17")
#> <div id="label1" class="section level2" number="0.1">
#> <h2><span class="header-section-number">0.1</span> Some other
#> header</h2>
#> </div>

# 3 lines
pandoc::pandoc_convert(text = "## Some other header {#label1}", to = "html4", args = c("--section-divs", "--number-sections"), version = "2.16.2")
#> <div id="label1" class="section level2" number="0.1">
#> <h2><span class="header-section-number">0.1</span> Some other header</h2>
#> </div>
```

<sup>Created on 2022-01-24 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>